scipy/_lib/meson.build

```
@@ -145,10 +145,31 @@ python_sources = [
   'array_api_compat/array_api_compat/torch/linalg.py',
 ]
 
-py3.install_sources(
+names = [
+  'array_api_compat/__init__.py',
+  'array_api_compat/_internal.py',
+  'array_api_compat/common/__init__.py',
+  'array_api_compat/common/_aliases.py',
+  'array_api_compat/common/_helpers.py',
+  'array_api_compat/common/_linalg.py',
+  'array_api_compat/common/_typing.py',
+  'array_api_compat/cupy/__init__.py',
+  'array_api_compat/cupy/_aliases.py',
+  'array_api_compat/cupy/_typing.py',
+  'array_api_compat/cupy/linalg.py',
+  'array_api_compat/numpy/__init__.py',
+  'array_api_compat/numpy/_aliases.py',
+  'array_api_compat/numpy/_typing.py',
+  'array_api_compat/numpy/linalg.py',
+  'array_api_compat/torch/__init__.py',
+  'array_api_compat/torch/_aliases.py',
+  'array_api_compat/torch/linalg.py',
+]
+
+install_data(
   python_sources,
-  preserve_path: true,
-  subdir: 'scipy/_lib'
+  install_dir: py3.get_install_dir() / 'scipy/_lib',
```

```suggestion
  install_dir: py3.get_install_dir() / 'scipy/_lib/array_api_compat',
```

Just realised that we can do this.